src/map/script.cpp

```diff
@@ -18270,25 +18279,38 @@ BUILDIN_FUNC(delmonsterdrop)
 		return SCRIPT_CMD_FAILURE;
 	}
 
-	if(mob) { //We got a valid monster, check for item drop on monster
-		unsigned char i;
-		for(i = 0; i < MAX_MOB_DROP_TOTAL; i++) {
-			if(mob->dropitem[i].nameid == item_id) {
-				mob->dropitem[i].nameid = 0;
-				mob->dropitem[i].rate = 0;
-				mob->dropitem[i].steal_protected = false;
-				mob->dropitem[i].randomopt_group = 0;
-				mob_reload_itemmob_data(); // Reload the mob search data stored in the item_data
-				script_pushint(st,1);
-				return SCRIPT_CMD_SUCCESS;
-			}
-		}
-		//No drop on that monster
-		script_pushint(st,0);
-	} else {
-		ShowWarning("delmonsterdrop: bad mob id given %d\n",script_getnum(st,2));
+	if (!mob) {
+		if (script_isstring(st, 2))
+			ShowWarning("delmonsterdrop: bad mob name given %s\n", script_getstr(st, 2));
+		else
+			ShowWarning("delmonsterdrop: bad mob id given %d\n", script_getnum(st, 2));
 		return SCRIPT_CMD_FAILURE;
 	}
```

Comment:
Can we combine that with the if/else right at the start?
The item db lookup is kind of useless, if we do not have a valid mob.
Also please change it to ShowError